app/models/user.rb

```diff
@@ -103,8 +103,35 @@ def decline_invite
     update(invited_by_id: nil)
   end
 
-  private
-    def email_without_domain
-      email.split('@').first
+  def process_unit(unit_params, user_is_guest)
+    if user_is_guest # if user is guest
+      if unit = self.unit # and has a unit
```

Comment:
what about `if user_is_guest && unit == self.unit`. Not sure if assignment is what you want to check here?